errors/errctx_test.go

```diff
@@ -0,0 +1,155 @@
+package errors
+
+import (
+	"context"
+	stdErrors "errors"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/errgo.v1"
+
+	"github.com/Scalingo/go-utils/logger"
+)
+
+func TestErrCtx_GetDeepestContext(t *testing.T) {
+	t.Run("It should return a nil context if the error is not wrapped with ErrCtx", func(t *testing.T) {
```

Comment:
Don't we want to never return nil (to prevent having to check for `nil` and return a default context? (`context.Background()`) ?

Reply:
Finally I'm wonder about this one.
In case we are not sure about the error type get could still call `RootCtx`. If no `ErrCtx` found we probably want to notice it to avoid the usage of a `context.Background()`.
If `RootCtx` returns nil, it is clear that there is no `ErrCtx` error wrapped in the given err.

WDYT?

Reply:
You're right, but it makes the usage trickier since it's not sure the user will now the `ctx` can be `nil`, it's really not a habit in Go

Reply:
True, what about returning an err?

Reply:
I've added a "not found" error in addition